backend_plugin/src/test/java/positives/evaluators/CorrectIndentationPositive.java

```diff
@@ -0,0 +1,280 @@
+	try {
+	    // Mock a document event with a single backslash placed at the beginning of the third line
+	    offset = doc.getLineOffset(2);
+	    assertFalse(mockUserInput(offset, SINGLE_SPACE));
+
+	    delayUserInput();
+
+	    // Mock a document event with a single backslash placed at the end of the second line
+	    offset = doc.getLineOffset(2) - 1;
+	    assertFalse(mockUserInput(offset, SINGLE_SPACE));
+	} catch (BadLocationException e) {
+	    // Should never get here
+	    fail("Should never see this error in: " + this.getClass().getSimpleName() + "::" + this.getClass().getName());
+	    e.printStackTrace();
+	}
+    }
+
+
+    /**
+     * Verifies that the evaluation function will not trigger when the user types " " + a character on line 1 and then adds a space at the start
+     * of line 2
+     */
+    @Test
+    public void spaceThenCharacterAtStartOfLine() {
+	try {
+	    // Mock a document event with a single space placed at the beginning of the first line
+	    offset = doc.getLineOffset(0);
+	    assertFalse(mockUserInput(offset, SINGLE_SPACE));
+	    delayUserInput();
+	    offset++;
+	    assertFalse(mockUserInput(offset, "b"));
+
+	    // Mock a document event with a single space placed at the start of the second line
+	    offset = doc.getLineOffset(1);
+	    assertFalse(mockUserInput(offset, SINGLE_SPACE));
+	} catch (BadLocationException e) {
+	    // Should never get here
+	    fail("Should never see this error in: " + this.getClass().getSimpleName() + "::" + this.getClass().getName());
+	    e.printStackTrace();
+	}
+    }
+
+    /**
+     * Verifies that the evaluation function will not trigger when the user removes space at the start then types a character on line 1 and then
+     * removes a space at the start of line 2
+     */
+    @Test
+    public void spaceRemovedThenCharacterAtStartOfLine() {
+	try {
+	    // Mock a document event with a single space placed at the beginning of the first line
+	    offset = doc.getLineOffset(0);
+	    assertFalse(mockUserSingleBackspace(offset));
+	    delayUserInput();
+	    offset++;
+	    assertFalse(mockUserInput(offset, "b"));
+
+	    // Mock a document event with a single space placed at the start of the second line
+	    offset = doc.getLineOffset(1);
+	    assertFalse(mockUserSingleBackspace(offset));
+	} catch (BadLocationException e) {
+	    // Should never get here
+	    fail("Should never see this error in: " + this.getClass().getSimpleName() + "::" + this.getClass().getName());
+	    e.printStackTrace();
+	}
+    }
+
+
+    ///////////////////////////////////
+    // Tests for adding / removing tabs
+    ///////////////////////////////////
+
+    /**
+     * Verifies that the evaluation function will not trigger when the user types " " anywhere but the start
+     * of a line
+     */
+    @Test
+    public void tabNotAtStartOfLineDown() {
+	try {
+	    // Mock a document event with a single space placed at the beginning of the first line
```

Comment:
It looks like some of the comments were copy/pasted from other methods, so it is a little bit confusing (i.e.this one should say "tab" instead of "space")